src/main/java/baritone/process/CraftingProcess.java

```diff
@@ -0,0 +1,430 @@
+import net.minecraft.init.Blocks;
+import net.minecraft.inventory.ClickType;
+import net.minecraft.inventory.ContainerPlayer;
+import net.minecraft.inventory.ContainerWorkbench;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.CraftingManager;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.network.play.client.CPacketPlaceRecipe;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+
+import java.util.*;
+
+public final class CraftingProcess extends BaritoneProcessHelper implements ICraftingProcess {
+    private int amount;
+    private IRecipe recipe;
+    private Goal goal;
+    private boolean placeCraftingTable = false;
+    private boolean clearToPush;
+    private List<BlockPos> knownLocations;
+
+    public CraftingProcess(Baritone baritone) {
+        super(baritone);
+    }
+
+    @Override
+    public boolean isActive() {
+        return recipe != null && amount >= 1;
+    }
+
+    @Override
+    public synchronized PathingCommand onTick(boolean calcFailed, boolean isSafeToCancel) {
+        if (calcFailed) {
+            logDirect("path calculation failed");
+            onLostControl();
+            return new PathingCommand(null, PathingCommandType.CANCEL_AND_SET_GOAL);
+        } else if (goal != null && !(goal instanceof GoalRunAway)) {
+            //we are pathing to a table and therefor have to wait.
+            baritone.getInputOverrideHandler().clearAllKeys();
+            if (goal.isInGoal(ctx.playerFeet())) {
+                rightClick();
+                if (ctx.player().openContainer instanceof ContainerWorkbench) {
+                    goal = null;
+                }
+            }
+            return new PathingCommand(goal, PathingCommandType.SET_GOAL_AND_PATH);
+        } else if (placeCraftingTable) {
+            placeCraftingTable();
+            return new PathingCommand(goal, PathingCommandType.SET_GOAL_AND_PATH);
+        } else {
+            //we no longer pathing so it's time to craft
+            try {
+                if (!canCraft(recipe,1) && amount > 0) {
+                    recipe = getCraftingRecipeForItem(recipe.getRecipeOutput().getItem());
+                    if (!canCraftInInventory(recipe) && !(ctx.player().openContainer instanceof ContainerWorkbench)) {
+                        pathToACraftingTable();
+                        return new PathingCommand(goal, PathingCommandType.SET_GOAL_AND_PATH);
+                    }
+                }
+                if (clearToPush) {
+                    moveItemsToCraftingGrid();
+                }
+                takeResultFromOutput();
+            } catch (Exception e) {
+                logDirect("Error! Did you close the crafting window while crafting process was still running?");
```

Comment:
We can easily detect a closed crafting screen so no need to assume it as the reason of any exception.

Reply:
i original added this try catch block to catch casting errors from generic Container to CraftingContainer, this however should no longer happen because the casting now checks beforehand if it can cast. (see getInputCount()) in theory the try catch block could be removed but i left it in should there be a edge case i didnt think of.